File: Castro-Lara-Daniel_Ivan/Procesos/files C++/signals.cpp
/*
    Sistemas Operativos
    Daniel Iván Castro Lara
    Programa en C++
	$ gcc -o procesos_v0_param procesos_v0_param.cpp
*/

#include <iostream>
#include <csignal>  // Para el manejo de señales
#include <unistd.h> // Para la función sleep

// Manejador de señales
void handler(int signum) {
    std::cout << "¡Señal recibida!" << std::endl;
}

int main() {
    // Asignar el manejador a la señal SIGUSR1
    std::signal(SIGUSR1, handler);

    std::cout << "Esperando señales... (envía SIGUSR1 para probar)" << std::endl;

    // Bucle infinito para mantener el programa en ejecución
    while (true) {
        sleep(1); // Evitar que el bucle consuma demasiados recursos
    }

    return 0;
}

File: Castro-Lara-Daniel_Ivan/Procesos/files Python/signals.py
"""
    Sistemas Operativos
    Daniel Iván Castro Lara
    Programa en Python 
"""

import signal
import time

# Manejador de la señal
def handler(signum, frame):
    print("¡Señal recibida!")

def main():
    # Configurar el manejador para la señal SIGUSR1
    signal.signal(signal.SIGUSR1, handler)
    print("Esperando señales... (envía SIGUSR1 para probar)")

    # Bucle infinito para mantener el programa en ejecución
    while True:
        time.sleep(1)  # Dormir para evitar el uso excesivo de CPU

if __name__ == "__main__":
    main()


"""
    1.- python <archivo>.py
    No se necesita compilar porque es interpretado.
"""
